ytypes/unmarshal.go

```diff
@@ -29,6 +29,14 @@ type UnmarshalOpt interface {
 	IsUnmarshalOpt()
 }
 
+// BestEffortUnmarshal is an unmarshal option that 'ignores' errors while unmarshalling,
+// and continues the unmarshaling process. A failing unmarshal will still fail, but it will
+// fail as a series of error reports.
+type BestEffortUnmarshal struct{}
```

Comment:
Perhaps make this unexported for now and export it after the functionality is ready in case we want to create another release before then?

Reply:
To enable early usage we've decided to keep this exported and just have the MVP functionality of accumulating errors in an error report.